### PR TITLE
Product (only create() func to connect with db and new .yaml + db configure)

### PR DESCRIPTION
**Pull request by MaximPolapa**

### CODE REVIEW

File: src/main/java/shippingService/dto/ProductDTO.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+import java.math.BigDecimal;
+import java.util.Objects;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class ProductDTO {
+    private Long productID;
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/dto/ProductDTO.java]: just id, not productID

File: src/main/java/shippingService/dto/UserDTO.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import javax.persistence.Id;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotNull;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class UserDTO {
+    @Id
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/dto/UserDTO.java]: revome @ Id

File: src/main/java/shippingService/dto/UserDTO.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    private Long id;
+    @Email
+    String email;
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/dto/UserDTO.java]: add private

File: src/main/java/shippingService/entity/Product.java [hunk 4]
```diff
@@ -17,9 +19,16 @@
 @Table(name = "product")
 public class Product {
     @Id
-    private Long id;
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long productID;
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/entity/Product.java]: just id, not productID

File: src/main/java/shippingService/entity/User.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "user")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Email
+    String email;
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/entity/User.java]: add private

File: src/main/java/shippingService/entity/User.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    @Column
+    String password;
+    @Column
+    String firstName;
+    @Column
+    String lastName;
+    @Column
+    String userRole;
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/entity/User.java]: Should be enum with Enumerated annotation

File: src/main/java/shippingService/resource/ProductResource.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("api/product")
+public class ProductResource {
+    @Autowired
+    private ProductService productService;
+
+    @PostMapping
+    public Product create(
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/resource/ProductResource.java]: return DTO here, not entity

File: src/main/java/shippingService/resource/ProductResource.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+            @Valid @RequestBody final Product product){
```
Comment (dmytro-krupa) [inline src/main/java/shippingService/resource/ProductResource.java]: ProductDTO, not Product

File: src/main/java/shippingService/entity/User.java [hunk 9]
```diff
@@ -26,7 +26,7 @@ public class User {
     @Column(name = "password")
     private String password;
 
-    @Column(name = "firstname")
+    @Column(name = "first_name")
     private String firstName;
 
     @Column(name = "lastName")
```
Comment (RIO489) [inline src/main/java/shippingService/entity/User.java]: lastName -> last_name

File: src/main/java/shippingService/resource/UserResource.java [hunk 10]
```diff
@@ -20,9 +20,34 @@ public class UserResource {
...
     }
+
+    @GetMapping("/{id}")
+    public UserDTO read(final @PathVariable Long id) {
+        return userService.read(id);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<UserDTO> update(final @PathVariable("id") Long id,
+                                          @RequestBody UserDTO userDTO) {
```
Comment (RIO489) [inline src/main/java/shippingService/resource/UserResource.java]: merge 34 and 35 rows in one straight row

File: src/main/java/shippingService/resource/ProductResource.java [hunk 11]
```diff
@@ -20,32 +20,33 @@ public class ProductResource {
...
 
     }
 
     @DeleteMapping("/{id}")
-    public String delete(final @PathVariable("id") Long id){
+    public String delete(final @PathVariable("id") Long id) {
         ProductDTO productDTO = productService.read(id);
         productService.delete(id);
-        return "Product:"+productDTO.toString();
+        return "Product:" + productDTO.toString();
```
Comment (RIO489) [inline src/main/java/shippingService/resource/ProductResource.java:44]: If user has "void delete".then product have to be the same

File: src/main/java/shippingService/service/impl/ProductServiceImpl.java [hunk 12]
```diff
@@ -42,12 +43,12 @@ public ProductDTO update(ProductDTO newProduct, Long id){
     }
```
Comment (RIO489) [inline src/main/java/shippingService/service/impl/ProductServiceImpl.java:38]: sh*tcode detected

Comment (RIO489) [reply src/main/java/shippingService/service/impl/ProductServiceImpl.java:38]: Please,refactor it properly

File: src/main/java/shippingService/service/impl/UserServiceImpl.java [hunk 13]
```diff
@@ -21,12 +23,34 @@ public class UserServiceImpl implements UserService {
...
 
+    @Override
+    public UserDTO update(UserDTO newUser, Long id) {
+        UserDTO oldUser = mapperUser.toDto(userRepository.findById(id).orElseThrow());
+        Long oldId = id;
+        oldUser = newUser;
+        oldUser.setId(id);
+        userRepository.save(mapperUser.toEntity(oldUser));
+        return oldUser;
+    }
```
Comment (RIO489) [inline src/main/java/shippingService/service/impl/UserServiceImpl.java:41]: sh&tcode detected
Please refactor it properly

**Review by RIO489 — APPROVED** [review]

**Pull request merged**